python/ray/tune/result_grid.py

```diff
@@ -258,6 +258,8 @@ def num_terminated(self):
 
     @staticmethod
     def _populate_exception(trial: Trial) -> Optional[Union[TuneError, RayTaskError]]:
+        if trial.num_failures_after_restore == 0:
+            return None
```

Comment:
Is this logic sufficient? If so can we just check `trial.status == "ERROR"`. instead?

Reply:
I think that makes sense, then we can get rid of the attribute. I'll update the PR in a bit